crypto/x509/x509_vfy.c

```diff
@@ -1728,55 +1684,54 @@ static int internal_verify(X509_STORE_CTX *ctx)
 {
     int n = sk_X509_num(ctx->chain) - 1;
     X509 *xi = sk_X509_value(ctx->chain, n);
-    X509 *xs;
+    X509 *xs = xi;
 
-    /*
-     * With DANE-verified bare public key TA signatures, it remains only to
-     * check the timestamps of the top certificate.  We report the issuer as
-     * NULL, since all we have is a bare key.
-     */
+    ctx->error_depth = n;
     if (ctx->bare_ta_signed) {
-        xs = xi;
+        /*
+         * With DANE-verified bare public key TA signatures,
+         * it remains only to check the timestamps of the top certificate.
```

Comment:
I've changed this to:
> With DANE-verified bare public key TA signatures,
> we only need to check the timestamps of the top certificate.

Reply:
This is still a little bit confusing because it somewhat implies, that we just check the timestamp of the top certificate and exit but we actually still need to verify the chain of signatures on the other certificates in the chain. What about `on the top certificate we check only the timestamps.` This changes the emphasis a little.

Reply:
I see - done.